Breaking change in controller-runtime: ctrl.NewWebhookManagedBy signature causes capsule-proxy build failure

<!--
Thanks for taking time reporting a Capsule bug!

-->

# Bug description

After updating dependencies, capsule-proxy fails to build due to a breaking change in controller-runtime.
# Root cause / impact
It appears that the signature of ctrl.NewWebhookManagedBy has changed in newer versions of controller-runtime. 
This prevents capsule-proxy from building when using newer versions of controller-runtime, breaking compatibility with updated dependency trees.

# Expected behavior

capsule-proxy should build successfully with supported versions of controller-runtime, or dependency constraints should prevent incompatible versions from being used.

# Logs
`Error: ../../../go/pkg/mod/github.com/projectcapsule/capsule@v0.12.1/api/v1beta1/tenant_webhook.go:18:34: not enough arguments in call to ctrl.NewWebhookManagedBy
    have (controllerruntime.Manager)
    want (manager.Manager, T)

make: *** [Makefile:76: ko-build-capsule-proxy] Error 1
`

--- a/api/v1beta1/tenant_webhook.go
+++ b/api/v1beta1/tenant_webhook.go
@@ -7,9 +7,10 @@ import (
 	"os"
 
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-func (in *Tenant) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func (in *Tenant) SetupWebhookWithManager(mgr manager.Manager) error {
 	certData, _ := os.ReadFile("/tmp/k8s-webhook-server/serving-certs/tls.crt")
 	if len(certData) == 0 {
 		return nil

